src/main/java/ro/secur/auth/service/UserService.java

```diff
@@ -88,8 +103,58 @@ public void changePassword(String username, ChangePasswordRequest request) {
         }
     }
 
+    public void forgotPassword(String email, HttpServletRequest request) {
+
+        String email1 = email.replace("{\"email\":{\"email\":\"", "");
+        String email2 = email1.replace("\"}}", "");
+        UserInfoEntity userInfoEntity = userInfoRepository.findByEmail(email2);
+        UserEntity userEntity = userRepository.findByUserInfoEntity(userInfoEntity);
+
+        if (userEntity != null) {
+            userEntity.setResetToken(UUID.randomUUID().toString());
+
+            save(userEntity);
+
+            String appUrl = request.getScheme() + "://" + request.getServerName() + ":" + request.getServerPort();
+
+            String token = userEntity.getResetToken();
+
+            SimpleMailMessage mailMessage = new SimpleMailMessage();
+            mailMessage.setTo(userEntity.getUserInfoEntity().getEmail());
+            mailMessage.setSubject("Complete Password Reset!");
+            mailMessage.setFrom("secur.app.20@gmail.com");
+            mailMessage.setText("To reset your password, click the link below:\n" + "http://localhost:3000/" + token + "/resetPassword");
```

Comment:
Can't we have the email details in properties file?

Reply:
lAici ai putea pune parametri de maill in app.prop. exemplul il poti vedea InJwtConfig.